src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonConcurrentQueueOfTConverter.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class JsonConcurrentQueueOfTConverter<TCollection, TElement> : JsonIEnumerableDefaultConverter<TCollection, TElement>
+        where TCollection : ConcurrentQueue<TElement>
+    {
+        protected override void Add(TElement value, ref ReadStack state)
+        {
+            ((TCollection)state.Current.ReturnValue!).Enqueue(value);
```

Comment:
We know  it's not `null` because we new it up in `CreateCollection`. This replaces whatever the user set in their object graph, pending any `ObjectCreationHandling` work.

Reply:
The ! should never result in a null on `state.Current.ReturnValue` since it is initialized in `CreateCollection` which is always called before `Add()`.

null collections are not handled by the converter.